scipy/signal/signaltools.py

```diff
@@ -4441,9 +4441,9 @@ def decimate(x, q, n=None, ftype='iir', axis=-1, zero_phase=True):
 
     else:  # IIR case
         if zero_phase:
-            y = filtfilt(b, a, x, axis=axis)
+            y = sosfiltfilt(b, a, x, axis=axis)
```

Comment:
Hmm, I see at least one CI failure that looks related:

```python
scipy/signal/signaltools.py:4446: in decimate
    y = sosfilt(b, a, x, axis=axis)
E   TypeError: sosfilt() got multiple values for argument 'axis'

```

Perhaps because there's a difference in call signatures between the functions, and `axis` is provided both positionally and by keyword? (I'm guessing a bit--I'm not super familiar with the signal functions)

Reply:
`b, a` is the TF-style output, you need to use `sos` output in the three branches above (I think it should be possible to request/get `sos`-form outputs directly from each)

Reply:
I wrote a small test just to look at the behavior of the function before and after change:

```python
from scipy.signal import decimate
import numpy as np

x = np.r_[np.zeros(0), 1, np.zeros(100)]
N = 2 ** 14
q = 5

y = decimate(x, q, ftype="fir")
```

And for both cases I had the same answer, thus I don't know why there is a failure like this in the tests.

Reply:
Yes sorry I started the PR, to be assigned to it, but I am still digging.

I have changed 
```python
b, a = firwin(n+1, 1. / q, window='hamming'), 1.
```
to 
```python

Reply:
Sorry I started the PR and pushed before I finish to write everything before I got assigned to it.

I pushed the final solution using `sos = tf2sos(b, a)` to be able to use sos filters.

I tested it on my local machine and everything passed.

Reply:
Unfortunately once in TF / `b, a` form the initial accuracy of the (internal) zpk representation is already lost, so using `tf2sos` is not as good as modifying each branch directly to specify the output type as `sos`